src/Umbraco.Web.UI.Client/src/views/common/infiniteeditors/datatypesettings/datatypesettings.controller.js

```diff
@@ -102,14 +102,27 @@
             vm.saveButtonState = "busy";
 
             var preValues = dataTypeHelper.createPreValueProps(vm.dataType.preValues);
-
+            
             dataTypeResource.save(vm.dataType, preValues, $scope.model.create).then(function(newDataType) {
                 $scope.model.dataType = newDataType;
                 vm.saveButtonState = "success";
 
                 if ($scope.model && $scope.model.submit) {
                     $scope.model.submit($scope.model);
                 }
+            }, function(err) {
+                vm.saveButtonState = "error";
+     
+                 if(err.status === 400) {
```

Comment:
@bergmania code looks fine but do we not have some nicer utility/helper stuff that does this kind of stuff already?

Reply:
I don't know. At least not anything that hooks in automatically, this was necessary to show the result of the server side validation